### PR TITLE
Support dynamic ids and import from another region

### PR DESCRIPTION
**Pull request by kryswisnaskas**
**Description of change**
Refactors change from https://github.com/adhocteam/Head-Start-TTADP/pull/77 to support dynamic ids as well as import from another region. Included "Grantee Name" as the expected column name in addition to "Grantee (distinct by row number)". Added test for a sample data from region 9.

**How to test**
Running locally:

- `yarn deps`
- `yarn db:migrate`
- `./node_modules/.bin/babel-node ./tools/importTTAPlanGoals.js -f R9GranteeTTAPlanTest.csv`
- verify that the data has been imported

To test using the original smartsheet file:

- Navigate to https://app.smartsheetgov.com/sheets/fjfHfQCg6mmvRP5ghV77wP9wQ5xg2P2H37QhF7Q1?view=grid
- Export the file to MS Excel or Google Sheets
- Replace smart quotes with non-smart quotes
- Replace "Anticipated Outcome" with "Goal" in the column names
- Save as comma delimited CSV, e.g. `R9GranteeTTAPlan.csv`
- Follow the above steps substituting the file name with `R9GranteeTTAPlan.csv`

**Issue(s)**
* https://github.com/HHS/Head-Start-TTADP/issues/208

**Checklist**

- [x] Meets issue criteria
- [x] Code tested
- [n/a] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [x] Documentation updated
    - API methods
    - Boundary and Data Flow Diagrams
    - [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions with the [Nygard template](https://github.com/joelparkerhenderson/architecture_decision_record/blob/master/adr_template_by_michael_nygard.md)
    - OSCAL templates completed when security controls are implemented or modified

### CODE REVIEW

File: tools/importPlanGoals.js [hunk 1]
```diff
@@ -83,29 +103,38 @@ export default async function importGoals(file) {
...
+                  const topicId = dbTopic.id;
                   if (roles) {
                     const rolesArr = roles.split(',');
-                    rolesArr.forEach((role) => {
+                    for await (const role of rolesArr) {
                       const trimmedRole = role.trim();
-                      const roleId = getItemId(cleanRoles, trimmedRole);
+                      let roleId;
+                      if (trimmedRole === 'GS') { // Special case for 'GS' since it's non-unique
+                        const [dbRole] = await Role.findOrCreate({ where: { name: trimmedRole, fullName: 'Grantee Specialist' } });
```
Comment (jasalisbury) [inline tools/importPlanGoals.js:119]: Are we assuming all `GS` roles from the CSV are "Grantee Specialists" and not "Grant Specialists"?

Comment (kryswisnaskas) [reply tools/importPlanGoals.js:119]: Yes, that is the assumption, which was confirmed with Ryan.

File: src/models/goal.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ module.exports = (sequelize, DataTypes) => {
     static associate(models) {
       Goal.belongsToMany(models.Topic, { through: models.TopicGoal, foreignKey: 'goalId', as: 'topics' });
       Goal.belongsToMany(models.Grantee, { through: models.GrantGoal, foreignKey: 'goalId', as: 'grantees' });
```
[the remark this answers is not in the document]
Comment (kryswisnaskas) [reply src/models/goal.js:15]: It's nice having the grantee in this table, even though the data is not entirely normalized.

[the remark this answers is not in the document]
Comment (kryswisnaskas) [reply src/models/goal.js:15]: We could remove in a different PR if it proves unnecessary.

**Review by jasalisbury — APPROVED** [review]
:+1:

**Pull request merged**